### PR TITLE
[android] Button to export all bookmarks

### PR DESCRIPTION
**Pull request by cyber-toad**
#7356 need to be merged first.
I stole all the code from #7357 :), thanks Roman

### CODE REVIEW

File: data/strings/strings.txt [hunk 1]
```diff
@@ -28803,3 +28803,48 @@
...
+    el = Εξαγωγή σελιδοδεικτών και διαδρομών
+    es = Exportar marcadores y pistas
+    et = Ekspordi järjehoidjad ja lood
+    eu = Esportatu laster-markak eta ibilbideak
+    fa = صادرات نشانک ها و آهنگ ها
+    fi = Kirjanmerkkien ja raitojen vienti
+    fr = Exporter les signets et les pistes
+    he = ייצוא סימניות ורצועות
+    hi = बुकमार्क और ट्रैक निर्यात करें
+    hu = Könyvjelzők és sávok exportálása
```
Comment (d4f5409d) [inline data/strings/strings.txt:28829]:
```suggestion
    hu = Könyvjelzők és útvonalak exportálása
```

**Review by d4f5409d — CHANGES_REQUESTED** [review]
One small change requested in 🇭🇺

File: android/app/src/main/java/app/organicmaps/bookmarks/BookmarkCategoriesAdapter.java [hunk 2]
```diff
@@ -26,6 +26,7 @@ public class BookmarkCategoriesAdapter extends BaseBookmarkCategoryAdapter<Recyc
   private final static int TYPE_CATEGORY_ITEM = 1;
   private final static int TYPE_ACTION_ADD = 2;
   private final static int TYPE_ACTION_IMPORT = 3;
+  private final static int TYPE_ACTION_EXPORT = 4;
```
Comment (biodranik) [inline android/app/src/main/java/app/organicmaps/bookmarks/BookmarkCategoriesAdapter.java:29]: EXPORT_ALL_AS_KMZ?

Users ask to export files in GPX too, although it's more about exporting only one file, we may also add exporting of all files as GPX in GPZ archive.

Or we'll add a format selection as a submenu after pressing the export button?

File: data/strings/strings.txt [hunk 3]
```diff
@@ -28803,3 +28803,48 @@
     vi = Âm lượng
     zh-Hans = 音量
     zh-Hant = 音量
+
+  [bookmarks_export]
+    tags = android
+    comment = Bookmark categories screen, button that opens folder selection dialog to export KML/KMZ/GPX/KMB files
```
Comment (biodranik) [inline data/strings/strings.txt:28809]: Why it should open a folder selection dialog? I would expect to see a standard system sharing dialog where users can choose the app where to export the file.

File: data/strings/strings.txt [hunk 4]
```diff
@@ -28803,3 +28803,48 @@
...
+    en = Export Bookmarks and Tracks
```
Comment (biodranik) [inline data/strings/strings.txt:28810]: Maybe 
```suggestion
    en = Export all Bookmarks and Tracks
```
?

File: android/app/src/main/java/app/organicmaps/bookmarks/BookmarkCategoriesAdapter.java (hunk 2 above)
Comment (cyber-toad) [reply android/app/src/main/java/app/organicmaps/bookmarks/BookmarkCategoriesAdapter.java:29]: Fixed

File: data/strings/strings.txt (hunk 3 above)
Comment (cyber-toad) [reply data/strings/strings.txt:28809]: It is a copy-paste issue in comment, it will be a share dialog

File: data/strings/strings.txt (hunk 1 above)
Comment (cyber-toad) [reply data/strings/strings.txt:28829]: Thanks for the suggestion, english text was slightly changed, could you please review updated translation

File: data/strings/strings.txt [hunk 5]
```diff
@@ -28803,3 +28803,50 @@
...
+    it = Esporta tutti i segnalibri e le tracce
+    ja = すべてのブックマークとトラックをエクスポートする
+    ko = 모든 북마크 및 트랙 내보내기
+    lt = Eksportuoti visas žymes ir takelius
+    mr = सर्व बुकमार्क आणि ट्रॅक निर्यात करा
+    nb = Eksporter alle bokmerker og spor
+    nl = Alle bladwijzers en tracks exporteren
+    pl = Eksportuj wszystkie zakładki i ścieżki
+    pt = Exportar todos os marcadores e faixas
+    pt-BR = Exportar todos os marcadores e trilhas
```
Comment (matheusgomesms) [inline data/strings/strings.txt:28841]:
```suggestion
    pt = Exportar todos os favoritos e trilhos
    pt-BR = Exportar todos os favoritos e trilhas
```

**Review by d4f5409d — CHANGES_REQUESTED** [review]
Hu

**Comment by cyber-toad** [comment]
> Hu

it looks like not the full comment was submitted?

**Review by biodranik — APPROVED** [review]
Thanks! Let's actively test it in the master.

@Jean-BaptisteC @rtsisyk can you please help with different test cases? Let's test that exported kmz files are properly opened in other software (Google Earth, umap, what else?)

**Comment by biodranik** [comment]
@kirylkaveryn can you please implement "Export all bookmarks and tracks" button on iOS?

**Pull request merged**